actors/builtin/market/deal.go

```diff
@@ -72,23 +72,11 @@ func (p *DealProposal) ProviderBalanceRequirement() abi.TokenAmount {
 }
 
 func (p *DealProposal) Cid() (cid.Cid, error) {
-	buf := new(bytes.Buffer)
-	if err := p.MarshalCBOR(buf); err != nil {
+	var buf bytes.Buffer
+	if err := p.MarshalCBOR(&buf); err != nil {
```

Comment:
Aside: the miner actor alone demonstrates three different ways of doing this. We had a problem early on where I propagated a broken pattern, though I think it was yet a fourth way. If this way is good, let's use it consistently.

- `var buf bytes.Buffer`
- `buf := bytes.Buffer{}`
- `buf := new(bytes.Buffer)`

Reply:
Well, they all work but I usually avoid the second case.

In this ase, I chose the `var` approach to make it clear that this doesn't escape to the heap. I guess `new` probably works just as well, but my brain always assumes "new means heap".

For now, I'll just revert this change.

Reply:
Me too, and the `var` approach is probably better. But the heterogeneity makes me have to think every time I see a different one.